SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/DormancyTests/DynamicActorAwakeAfterDormantChangePropertyTest.cpp

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "DynamicActorAwakeAfterDormantChangePropertyTest.h"
+
+#include "DormancyTestActor.h"
+
+// This test checks whether changes on a dormant actor are replicated when the actor becomes awake.
+
+ADynamicActorAwakeAfterDormantChangePropertyTest::ADynamicActorAwakeAfterDormantChangePropertyTest()
+{
+	Author = "Matthew Sandford";
+}
+
+void ADynamicActorAwakeAfterDormantChangePropertyTest::PrepareTest()
+{
+	Super::PrepareTest();
+
+	// Step 1 - Spawn dormancy actor and change NetDormancy to DORM_DormantAll
+	AddStep(TEXT("ServerSpawnDormancyActor"), FWorkerDefinition::Server(1), nullptr, [this]() {
+		AActor* Actor = CreateDormancyTestActor();
+		Actor->SetNetDormancy(DORM_DormantAll);
+		FinishStep();
+	});
+
+	// Step 2 - Client check NetDormancy is DORM_DormantAll
+	AddStep(
+		TEXT("ClientCheckDormancyAndRepProperty"), FWorkerDefinition::AllClients, nullptr, nullptr,
+		[this](float DeltaTime) {
+			CheckDormancyAndRepProperty(DORM_DormantAll, 0);
```

Comment:
Hm, maybe check if the actor is in the world of the client? The loop for `CheckDormancyAndRepProperty` would pass if it had 0 actors in the client world, right?